From 26b92aec405f723d920fb5c0582d8bd32dde51fe Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Tue, 13 Oct 2026 05:35:56 +0200
Subject: [PATCH] avcodec/mpegvideo_enc: return the error code of
 encode_picture()

The error was replaced by -1, shown to the user as EPERM.

Fixes: PUUZOCCqIPbY
Fixes: wrong error code
Wrong error code Replicated through UnModified FFmpeg with ffmpeg command line
Found during triage of the review of https://code.ffmpeg.org/FFmpeg/FFmpeg/pulls/24758

(cherry picked from commit 26b92aec405f723d920fb5c0582d8bd32dde51fe)
---
 libavcodec/mpegvideo_enc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavcodec/mpegvideo_enc.c b/libavcodec/mpegvideo_enc.c
index 9f5ec9f6504..c9da0326164 100644
--- a/libavcodec/mpegvideo_enc.c
+++ b/libavcodec/mpegvideo_enc.c
@@ -1969,7 +1969,7 @@ vbv_retry:
             pkt->size = avctx->internal->byte_buffer_size;
         }
         if (ret < 0)
-            return -1;
+            return ret;
 
         frame_end(m);
 
-- 
2.53.0

